### ISSUE TITLE
how to deal with shared siamese during bp

### ISSUE THREAD

**Issue by yupup** [bertinetto/siamese-fc#34]
Hi Bertinetto,

When you define the shared siamese net, in the backpropagation process.
When it reach the first BatchNorm, after the norm, it will set obj.moments=[] (See matconvnet/matlab/+dagnnn/BatchNorm.m, the backward() function).
Then the second BatchNorm has an empty obj.moments, it will cause error.

How did you deal with that, did you modified the code somehow?

**Comment by yupup** [comment]
P.S.: I'm using matconvnet beta25,    I saw the changelog (http://www.vlfeat.org/matconvnet/about/#about-matconvnet), it changes the behavior of that since beta21,  this obj.moments=[] is added. 

Then the question I guess would be, how to adapt to this new change in your code? Maybe some config?
